core/src/main/java/org/elasticsearch/index/query/SpanContainingQueryBuilder.java

```diff
@@ -92,9 +84,16 @@ protected Query doToQuery(QueryParseContext parseContext) throws IOException {
 
     @Override
     public QueryValidationException validate() {
-        QueryValidationException validationExceptions = validateInnerQuery(big, null);
-        validationExceptions = validateInnerQuery(little, validationExceptions);
-        return validationExceptions;
+        QueryValidationException validationException = null;
+        if (big == null) {
+            validationException = addValidationError("inner clause [big] cannot be null.", validationException);
+        }
+        if (little == null) {
+            validationException = addValidationError("inner clause [little] cannot be null.", validationException);
+        }
+        validationException = validateInnerQuery(big, validationException);
+        validationException = validateInnerQuery(little, validationException);
```

Comment:
Good suggestions, only problem I can think of right now is that we then have to pass an additional message down that we use in case of null, and at the moment I like that we see every potential problem thats's covered with the current query class also in the code here. Will see how that looks like.